### ISSUE TITLE
Browser struggling with live logs of submissions on many tasks

### ISSUE THREAD

**Issue by lbeziaud** [codalab/codabench#809]
Hi,

I get inconsistent or incomplete error reporting. With a benchmark created from [bundle.zip](https://github.com/codalab/codabench/files/11248254/bundle.zip) and submission as [cheating_kit.zip](https://github.com/codalab/codabench/files/11248280/cheating_kit.zip), I get:
1. two "different" errors (line missing) although I should get the same error for each task;
2. the errors seems to be truncated;
3. the log tab is empty (no stdout, stderr).

Edit: The submission is evaluated against 27 tasks, which is likely to be the source of the issue. After refreshing the page, the detail tab (screenshot 3) is fixed (errors appear in stderr tab and are not truncated). However opening the tab takes ~20 seconds and gives me a "This page is slowing down Firefox" due to the 216 xhr requests). Might this be worth investigating?

![Screen Shot 2023-04-17 at 11 04 27](https://user-images.githubusercontent.com/4895292/232438111-fdc66c36-2c2b-490a-b4cd-182b3744f46c.png)

![Screen Shot 2023-04-17 at 11 04 32](https://user-images.githubusercontent.com/4895292/232438105-1881e016-acec-4444-8923-07eb2d5067a4.png)

![Screen Shot 2023-04-17 at 11 10 18](https://user-images.githubusercontent.com/4895292/232439266-3c54e311-1a1c-4231-8509-ef9fff09631d.png)

**Comment by Didayolo** [comment]
So from my understanding, the browser had difficulties with your submission on many tasks, but there were actually no bug?

**Comment by lbeziaud** [comment]
Well there is a bug, but (I think) it is due to the browser struggling with many tasks. The bug concerns the live log output, during the evaluation of the submission. Once the submission is done being evaluated, the log output is fine. It is technically a bug, however it's not really crucial.

The issue of the browser struggling with "many" tasks might be better of in a standalone issue. Showing the details of a submission takes a really long time since there is a lot of calls for each tasks. 
For example with 12 tasks it takes around 6 seconds to load. Looking at the network calls, a lot of requests are made for each task (scoring_stderr.txt, scoring_stdout.txt, scoring_ingestion_stderr.txt, scoring_ingestion_stdout.txt,  prediction_ingestion_stdout.txt, prediction_ingestion_stderr.txt, prediction_stdout.txt, prediction_sterr.txt). 
(1) They seem to be made synchronously which makes for a long waiting time before the window appear. This could be worth optimizing for (handling of multiple tasks per submission is a really nice feature!). (By the way all the requested files in this example are empty).
(2) This example is only a scoring task, so getting scoring_ingestion, prediction_ingestion, and prediction files is not necessary. That would save a lot of requests!

Feel free to close of open another issue :)

**Comment by Didayolo** [comment]
Related:

- #1031
